packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/lib/feature_counter/blocs/counter_bloc.dart

```diff
@@ -50,18 +50,25 @@ class CounterBloc extends $CounterBloc {
   @override
   Stream<int> _mapToCountState() => Rx.merge<Result<int>>([
         // On increment.
-        _$incrementEvent
-            .switchMap((_) => _repository.increment().asResultStream()),
+        _$incrementEvent.switchMap((_) => _repository
```

Comment:
@RomanovaPrime, I would turn the `Rx.merge<Result<int>>` to `Rx.merge<Result<Count>>`, and right after .whereSuccess() I would add a mapper to get the `int` from the `Counter`. In this way, the code will look much more simple.

Reply:
Edited.